### ISSUE TITLE
Link A Link B Link C removal

### ISSUE THREAD

**Issue by webchun** [joomla/joomla-cms#30280]
### Steps to reproduce the issue
I have never used Link A Link B Link C when creating content, and when I really need it I think now we can use Custom Field. 
Why don't we just remove it? I don't see any benefit to keep it and it might reduce confusion for new users/adopters. It also makes Joomla is simpler and easier to use.

![screen shot 2020-08-05 at 04 17 06](https://issues.joomla.org/uploads/1/59cd4063c4b561679d73c359f71e0252.jpg)

**Comment by brianteeman** [comment]
Just because you have never used it doesn't mean that other people do not. If it is removed what happens to the existing sitess that are using it

**Comment by astridx** [comment]
Brian is right. Simply removing it would not be good. On the other hand, it is good to clean up every now and then. Perhaps this is a compromise: we could mark the fields depreciated and only show them in the backend if they are filled. In this way, they can be changed but no new use is possible.

**Comment by coolcat-creations** [comment]
Can't we migrate the fields to custom fields? Having a custom field Group Images and Links that show exactly that but can be deactivated?

**Comment by brianteeman** [comment]
@coolcat-creations not going to be easy to do at all. I did have a PR to separate the "show images/links" param into "show images" and "show links" but it didnt gain any traction

**Comment by HLeithner** [comment]
The next possible release for such change is J5, the only thing you can do is to "hide" them but I don't see any benefit for this. 

You can move Links to it's own tab and hide it per default for new installations.

**Comment by coolcat-creations** [comment]
I would welcome such change. The less unneeded stuff displayed, the better. But it would be great to have a link Custom field were you can enter an external or internal link. Internal link like file, menu item or article. Should I create a feature request?

**Comment by webchun** [comment]
> Brian is right. Simply removing it would not be good. On the other hand, it is good to clean up every now and then. Perhaps this is a compromise: we could mark the fields depreciated and only show them in the backend if they are filled. In this way, they can be changed but no new use is possible.

This is the best option we have I think, I believe this small change can improve the UX for new adopters.
We probably will also get feedback, to know how many people actually use it before it totally be removed it in J5.

**Comment by 4digitalworld** [comment]
> 
> 
> @coolcat-creations not going to be easy to do at all. I did have a PR to separate the "show images/links" param into "show images" and "show links" but it didnt gain any traction

i think a good solution would be to create a "**migrator**" to transfer values from this field.

the same can be done with the "Images" field.

Now this tab is a duplicate of those fields that are in @Joomla.

**Comment by brianteeman** [comment]
Good luck creating that

**Comment by obuisard** [comment]
There is more to it than just removing the fields and use custom fields instead.
If you want to remove Links A, B and C, the custom field of type URL also needs to be able to store a URL text and a target window, parameters which are desperately missing from the custom field. Loosing that info will not satisfy users.

**Comment by obuisard** [comment]
While I am at it: getting links from the content table is 'straightforward' (through the 'urls' table column) while getting the custom fields for an article requires not so straightforward database requests. Basically, by using custom fields rather than Links A, B and C, we will be deteriorating performance.

**Comment by Llewellynvdm** [comment]
The reality is it is not confusing, since those who know how to do the same with custom fields aren't new users, but has at least been around the block a few times. So I agree, if it is not broke, then don't fix it :+1:

**Comment by ghost** [comment]
Link A Link B Link C should be removed. They are confusing and duplicate

**Comment by brianteeman** [comment]
@chnnst and how do you propose handling upgrades where they are being used?

**Comment by ghost** [comment]
Link A Link B Link C should be removed. They are confusing and duplicate, no one should use them

**Comment by brianteeman** [comment]
Stop and think for a second before posting and liking your own comments.

No matter what you think of these fields they exist in joomla 3 and people will have used them so if you remove them you have to have a means to handle that data on a site upgrade otherwise you will lose data and your site will be broken.

Think of everyone using joomla not just yourself

**Comment by ghost** [comment]
Theus who use confusing and duplicate features can upgrade using extension or later if there will be need in j 4.1.

We need to ship j4 as soon as we can, with no need to introduce confusing and duplicate features.

Link A Link B Link C should be removed. They are confusing and duplicate, no one should use them.

**Comment by HLeithner** [comment]
repeating the same message over and over again doesn't make it true, the last guy that did that got blocked on twitter, facebook, youtube, instagram and his own webshop. So please stop wasting our time with the same message.

**Comment by ghost** [comment]
HLeithner joomla 4 is without release date and delayed. We need to ship joomla 4 as soon as we can, why you want to introduce confusing and duplicate features in joomla 4 ?

If you use joomla 3 with Link A Link B Link C you can upgrade later if there will be need using extension or in j 4.1 or j 5

Why not to do this?

**Comment by ghost** [comment]
As a new user, I see it like @chnnst. 
Why not at least use https://github.com/joomla/joomla-cms/issues/30280#issuecomment-669057675

**Comment by brianteeman** [comment]
> Why not to do this?

Please think before you post. If you did you would realise why you are talking rubbish. This is NOT new. Have you ever used Joomla?

**Comment by HLeithner** [comment]
@chnnst we do exactly this, we don't delay j4 and keep the behavior that exists at least 2011?

**Comment by ghost** [comment]
HLeithner in my opinion introducing confusing and duplicate features in joomla 4 is bad design, but beter to discuss it couse it seems we have different opinions

https://github.com/joomla/joomla-cms/discussions/32179

**Comment by ghost** [comment]
> @chnnst we do exactly this, we don't delay j4 and keep the behavior that exists at least 2011?

Who is we?

**Comment by HLeithner** [comment]
you are so a funny person and you don't understand that this isn't an introduction because it already exsists. So I look this topic for now and can be unlocked at a later stage in the j5 dev process if needed.

**Comment by HLeithner** [comment]
@agi-code https://github.com/joomla/joomla-cms/commit/038d5c8e4361487d388eebb9acdefc13bb06efdf is the related commit

**Comment by MacJoom** [comment]
The links will not be removed.
